src/webui/webapplication.cpp

```diff
@@ -893,98 +915,265 @@ void WebApplication::action_command_setCategory()
+    const QString destPath = request().posts["destPath"].trimmed();
+    QString error;
+    if (!RSS::Session::instance()->moveItem(itemPath, destPath, &error)) {
+        status(409, "Conflict");
+        print(error, Http::CONTENT_TYPE_TXT);
+    }
+}
+
+void WebApplication::getRSSItems()
+{
+    print(QJsonDocument(
+              RSS::Session::instance()->rootFolder()->toJsonValue(
+                  parseBool(request().gets["withData"], false)
+              ).toObject()).toJson()
+            , Http::CONTENT_TYPE_JSON);
+}
+
+void WebApplication::setRSSRule()
+{
+    CHECK_PARAMETERS("ruleName" << "ruleDef");
+
+    const RSS::AutoDownloadRule rule {
+        RSS::AutoDownloadRule::fromJsonObject(
+                    QJsonDocument::fromJson(request().posts["ruleDef"].trimmed().toUtf8()).object()
+                , request().posts["ruleName"].trimmed())
+    };
+
+    RSS::AutoDownloader::instance()->insertRule(rule);
+}
+
+void WebApplication::renameRSSRule()
+{
+    CHECK_PARAMETERS("ruleName" << "newRuleName");
+
+    const QString ruleName = request().posts["ruleName"].trimmed();
+    const QString newRuleName = request().posts["newRuleName"].trimmed();
+    RSS::AutoDownloader::instance()->renameRule(ruleName, newRuleName);
+}
+
+void WebApplication::removeRSSRule()
+{
+    CHECK_PARAMETERS("ruleName");
+
+    const QString ruleName = request().posts["ruleName"].trimmed();
+    RSS::AutoDownloader::instance()->removeRule(ruleName);
+}
+
+void WebApplication::getRSSRules()
+{
+    QJsonObject jsonObj;
+    foreach (auto rule, RSS::AutoDownloader::instance()->rules())
+        jsonObj.insert(rule.name(), rule.toJsonObject());
+
+    print(QJsonDocument(jsonObj).toJson(), Http::CONTENT_TYPE_JSON);
+}
+
+void WebApplication::getDefaultSavePath()
 {
-    CHECK_URI(0);
     print(BitTorrent::Session::instance()->defaultSavePath());
 }
 
-bool WebApplication::isPublicScope()
+bool WebApplication::isAPIMethod(const QString &methodName) const
 {
-    return (scope_ == DEFAULT_SCOPE || scope_ == VERSION_INFO);
+    return (m_privateAPIs.contains(methodName)
+            || m_publicAPIs.contains(methodName));
+}
+
+bool WebApplication::isPublicAPIMethod(const QString &methodName) const
+{
+    return m_publicAPIs.contains(methodName);
+}
+
+QString WebApplication::getParam(const QString &name) const
+{
+    if (request().method == Http::METHOD_GET) {
+        if ((name == QLatin1String("hash")) && !m_hashCompatParam.isEmpty())
+            return m_hashCompatParam;
+        return request().gets[name];
+    }
+
+    return request().posts[name];
 }
 
 void WebApplication::doProcessRequest()
 {
-    scope_ = DEFAULT_SCOPE;
-    action_ = DEFAULT_ACTION;
+    QString methodName;
+
+    if (request().path.startsWith(PATH_PREFIX_API)) {
+        methodName = request().path.mid(PATH_PREFIX_API.size());
+    }
+    else if ((methodName = APICompatMapping1[request().path]).isEmpty()) {
```

Comment:
@glassez 
Can you please add a `warning` http header (via `ResponseBuilder::header()`) when the client uses the old api path?
see: https://stackoverflow.com/a/29623798